src/Symfony/Component/VarDumper/Caster/ResourceCaster.php

```diff
@@ -88,4 +107,24 @@ public static function castOpensslX509($h, array $a, Stub $stub, bool $isNested)
 
         return $a;
     }
+
+    public static function castOpensslAsymmetricKey(\OpenSSLAsymmetricKey $key, array $a, Stub $stub, bool $isNested): array
```

Comment:
I would put them in an OpenSSLCaster instead of ResourceCaster, as they are not resources

Reply:
Should a new `castOpensslX509()` method also go in a new caster and `ResourceCaster:: castOpensslX509()` would call it?